### ISSUE TITLE
[bitnami/mysql:8.3.0-debian-12-r8] [ERROR] [MY-000067] [Server] unknown variable 'master-info-repository=TABLE'

### ISSUE THREAD

**Issue by Adsryen** [bitnami/containers#65677]
### Name and Version

bitnami/mysql:8.3.0-debian-12-r8

### What architecture are you using?

amd64

### What steps will reproduce the bug?

Either way I pass https://github.com/bitnami/containers/blob/main/bitnami/mysql/docker-compose-replication.yml
When deploying the files here, it is still a manually written compose. The master library can always start normally, but the slave library will always report this error.

unknown variable 'master-info-repository=TABLE'

### What do you see instead?

This is done via this file https://github.com/bitnami/containers/blob/main/bitnami/mysql/docker-compose-replication.yml
Deployed docker logs:
```
mysql 15:08:28.21 INFO  ==> 
mysql 15:08:28.21 INFO  ==> Welcome to the Bitnami mysql container
mysql 15:08:28.22 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
mysql 15:08:28.22 INFO  ==> Submit issues and feature requests at https://github.com/bitnami/containers/issues
mysql 15:08:28.22 INFO  ==> Upgrade to Tanzu Application Catalog for production environments to access custom-configured and pre-packaged software components. Gain enhanced features, including Software Bill of Materials (SBOM), CVE scan result reports, and VEX documents. To learn more, visit https://bitnami.com/enterprise
mysql 15:08:28.22 INFO  ==> 
mysql 15:08:28.23 INFO  ==> ** Starting MySQL setup **
mysql 15:08:28.26 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mysql 15:08:28.26 INFO  ==> Initializing mysql database
mysql 15:08:28.28 INFO  ==> Updating 'my.cnf' with custom configuration
mysql 15:08:28.29 INFO  ==> Setting user option
mysql 15:08:28.30 INFO  ==> Setting slow_query_log option
mysql 15:08:28.31 INFO  ==> Setting long_query_time option
mysql 15:08:28.32 INFO  ==> Installing database
2024-04-24T15:08:28.350886Z 0 [System] [MY-015017] [Server] MySQL Server Initialization - start.
2024-04-24T15:08:28.353544Z 0 [Warning] [MY-011070] [Server] 'binlog_format' is deprecated and will be removed in a future release.
2024-04-24T15:08:28.353567Z 0 [Warning] [MY-011068] [Server] The syntax 'log_slave_updates' is deprecated and will be removed in a future release. Please use log_replica_updates instead.
2024-04-24T15:08:28.353740Z 0 [System] [MY-013169] [Server] /opt/bitnami/mysql/bin/mysqld (mysqld 8.3.0) initializing of server in progress as process 55
2024-04-24T15:08:28.363957Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-04-24T15:08:28.769545Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
2024-04-24T15:08:31.220467Z 0 [ERROR] [MY-000067] [Server] unknown variable 'master-info-repository=TABLE'.
2024-04-24T15:08:31.221502Z 0 [ERROR] [MY-013236] [Server] The designated data directory /bitnami/mysql/data/ is unusable. You can remove all files that the server added to it.
2024-04-24T15:08:31.221572Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-04-24T15:08:33.482006Z 0 [System] [MY-015018] [Server] MySQL Server Initialization - end.
```

This is the custom compose and log:
```
version: '3'

services:
  mysql-slave-2:
    image: bitnami/mysql:8.3.0-debian-12-r8
    container_name: mysql-slave-2
    ports:
      - 3302:3306
    environment:
      - ALLOW_EMPTY_PASSWORD=no  
      - MYSQL_AUTHENTICATION_PLUGIN=mysql_native_password  
      - MYSQL_ROOT_USER=root  
      - MYSQL_ROOT_PASSWORD=hello
      - MYSQL_USER=ehr 
      - MYSQL_PASSWORD=hello
      - MYSQL_DATABASE=ehr  
      - MYSQL_MASTER_HOST=192.168.1.242 
      - MYSQL_MASTER_PORT_NUMBER=3300  
      - MYSQL_MASTER_ROOT_USER=root  
      - MYSQL_MASTER_ROOT_PASSWORD=hello
      - MYSQL_MASTER_DELAY=0
      - MYSQL_REPLICATION_USER=slave  
      - MYSQL_REPLICATION_PASSWORD=hello
      - MYSQL_PORT_NUMBER=3306  
      - MYSQL_REPLICATION_MODE=slave 
      - MYSQL_REPLICATION_SLAVE_DUMP=false 
      - MYSQL_STARTUP_WAIT_RETRIES=300 
      - MYSQL_STARTUP_WAIT_SLEEP_TIME=2 
      - MYSQL_ENABLE_SLOW_QUERY=1 
      - MYSQL_LONG_QUERY_TIME=10.0  
    volumes:
      - /root/docker_data/bitnami/mysql-slave-2/data:/bitnami/mysql/data
    healthcheck:
      test: ['CMD', '/opt/bitnami/scripts/mysql/healthcheck.sh']
      interval: 15s
      timeout: 5s
      retries: 6
```

```
mysql 14:49:49.07 INFO  ==> 
mysql 14:49:49.07 INFO  ==> Welcome to the Bitnami mysql container
mysql 14:49:49.07 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
mysql 14:49:49.08 INFO  ==> Submit issues and feature requests at https://github.com/bitnami/containers/issues
mysql 14:49:49.08 INFO  ==> Upgrade to Tanzu Application Catalog for production environments to access custom-configured and pre-packaged software components. Gain enhanced features, including Software Bill of Materials (SBOM), CVE scan result reports, and VEX documents. To learn more, visit https://bitnami.com/enterprise
mysql 14:49:49.08 INFO  ==> 
mysql 14:49:49.09 INFO  ==> ** Starting MySQL setup **
mysql 14:49:49.11 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mysql 14:49:49.12 INFO  ==> Initializing mysql database
mysql 14:49:49.14 INFO  ==> Updating 'my.cnf' with custom configuration
mysql 14:49:49.15 INFO  ==> Setting user option
mysql 14:49:49.16 INFO  ==> Setting port option
mysql 14:49:49.18 INFO  ==> Setting default_authentication_plugin option
mysql 14:49:49.19 INFO  ==> Setting slow_query_log option
mysql 14:49:49.19 INFO  ==> Setting long_query_time option
mysql 14:49:49.20 INFO  ==> Using persisted data
mysql 14:49:49.24 INFO  ==> Running mysql_upgrade
mysql 14:49:49.26 INFO  ==> Starting mysql in background
2024-04-24T14:49:49.282339Z 0 [System] [MY-015015] [Server] MySQL Server - start.
2024-04-24T14:49:49.566515Z 0 [Warning] [MY-011070] [Server] 'binlog_format' is deprecated and will be removed in a future release.
2024-04-24T14:49:49.566550Z 0 [Warning] [MY-011068] [Server] The syntax 'log_slave_updates' is deprecated and will be removed in a future release. Please use log_replica_updates instead.
2024-04-24T14:49:49.566570Z 0 [Warning] [MY-011068] [Server] The syntax 'skip_slave_start' is deprecated and will be removed in a future release. Please use skip_replica_start instead.
2024-04-24T14:49:49.566857Z 0 [Warning] [MY-010918] [Server] 'default_authentication_plugin' is deprecated and will be removed in a future release. Please use authentication_policy instead.
2024-04-24T14:49:49.566893Z 0 [System] [MY-010116] [Server] /opt/bitnami/mysql/bin/mysqld (mysqld 8.3.0) starting as process 94
2024-04-24T14:49:49.580810Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-04-24T14:49:49.747114Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
mysqld: Table 'mysql.plugin' doesn't exist
2024-04-24T14:49:49.903073Z 0 [ERROR] [MY-010735] [Server] Could not open the mysql.plugin table. Please perform the MySQL upgrade procedure.
2024-04-24T14:49:49.903580Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-24T14:49:49.903955Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-24T14:49:49.904392Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-24T14:49:49.904963Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-24T14:49:49.905273Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-24T14:49:49.906185Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-24T14:49:49.906462Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-24T14:49:49.964564Z 4 [System] [MY-013381] [Server] Server upgrade from '80300' to '80300' started.
2024-04-24T14:49:55.023337Z 4 [System] [MY-013381] [Server] Server upgrade from '80300' to '80300' completed.
2024-04-24T14:49:55.280206Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2024-04-24T14:49:55.280342Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2024-04-24T14:49:55.289643Z 0 [ERROR] [MY-000067] [Server] unknown variable 'master-info-repository=TABLE'.
2024-04-24T14:49:55.291373Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-04-24T14:49:57.259142Z 0 [System] [MY-010910] [Server] /opt/bitnami/mysql/bin/mysqld: Shutdown complete (mysqld 8.3.0)  Source distribution.
2024-04-24T14:49:57.259253Z 0 [System] [MY-015016] [Server] MySQL Server - end.
mysql 14:59:51.70 ERROR ==> MySQL failed to start
```

**Comment by javsalgar** [comment]
Hi!

I'm afraid we do not know Chinese, do you mind translating the issue to english?

**Comment by Adsryen** [comment]
> Hi!
> 
> I'm afraid we do not know Chinese, do you mind translating the issue to English?

I have translated it into English. I hope my description did not misinterpret my original Chinese meaning. I hope you can understand it.

**Comment by dgomezleon** [comment]
Hi @Adsryen 

> This is the custom compose and log:
> 
> ```
> version: '3'
> 
> services:
>   mysql-slave-2:
>     image: bitnami/mysql:8.3.0-debian-12-r8
>     container_name: mysql-slave-2
>     ports:
>       - 3302:3306
>     environment:
>       - ALLOW_EMPTY_PASSWORD=no  
>       - MYSQL_AUTHENTICATION_PLUGIN=mysql_native_password  
>       - MYSQL_ROOT_USER=root  
>       - MYSQL_ROOT_PASSWORD=hello
>       - MYSQL_USER=ehr 
>       - MYSQL_PASSWORD=hello
>       - MYSQL_DATABASE=ehr  
>       - MYSQL_MASTER_HOST=192.168.1.242 
>       - MYSQL_MASTER_PORT_NUMBER=3300  
>       - MYSQL_MASTER_ROOT_USER=root  
>       - MYSQL_MASTER_ROOT_PASSWORD=hello
>       - MYSQL_MASTER_DELAY=0
>       - MYSQL_REPLICATION_USER=slave  
>       - MYSQL_REPLICATION_PASSWORD=hello
>       - MYSQL_PORT_NUMBER=3306  
>       - MYSQL_REPLICATION_MODE=slave 
>       - MYSQL_REPLICATION_SLAVE_DUMP=false 
>       - MYSQL_STARTUP_WAIT_RETRIES=300 
>       - MYSQL_STARTUP_WAIT_SLEEP_TIME=2 
>       - MYSQL_ENABLE_SLOW_QUERY=1 
>       - MYSQL_LONG_QUERY_TIME=10.0  
>     volumes:
>       - /root/docker_data/bitnami/mysql-slave-2/data:/bitnami/mysql/data
>     healthcheck:
>       test: ['CMD', '/opt/bitnami/scripts/mysql/healthcheck.sh']
>       interval: 15s
>       timeout: 5s
>       retries: 6
> ```


You need to add the `mysql-master` part to this docker-compose to make it work. As you mentioned, you can use the [original one](https://github.com/bitnami/containers/blob/main/bitnami/mysql/docker-compose-replication.yml) as example, and edit values in order to add more slaves or to add environment variables.


I hope it helps

**Comment by mhkarimi1383** [comment]
Hi
I'm having the same problem 

Here is my complete log

```
mysql 17:32:00.38 INFO  ==> 
mysql 17:32:00.38 INFO  ==> Welcome to the Bitnami mysql container
mysql 17:32:00.39 INFO  ==> Subscribe to project updates by watching https://github.com/bitnami/containers
mysql 17:32:00.39 INFO  ==> Submit issues and feature requests at https://github.com/bitnami/containers/issues
mysql 17:32:00.40 INFO  ==> Upgrade to Tanzu Application Catalog for production environments to access custom-configured and pre-packaged software components. Gain enhanced features, including Software Bill of Materials (SBOM), CVE scan result reports, and VEX documents. To learn more, visit https://bitnami.com/enterprise
mysql 17:32:00.40 INFO  ==> 
mysql 17:32:00.41 INFO  ==> ** Starting MySQL setup **
mysql 17:32:00.45 INFO  ==> Validating settings in MYSQL_*/MARIADB_* env vars
mysql 17:32:00.46 INFO  ==> Initializing mysql database
mysql 17:32:00.49 INFO  ==> Updating 'my.cnf' with custom configuration
mysql 17:32:00.50 INFO  ==> Setting user option
mysql 17:32:00.52 INFO  ==> Setting slow_query_log option
mysql 17:32:00.54 INFO  ==> Setting long_query_time option
mysql 17:32:00.56 INFO  ==> Using persisted data
mysql 17:32:00.61 INFO  ==> Running mysql_upgrade
mysql 17:32:00.63 INFO  ==> Starting mysql in background
2024-04-28T14:02:00.665103Z 0 [System] [MY-015015] [Server] MySQL Server - start.
2024-04-28T14:02:01.081452Z 0 [Warning] [MY-011070] [Server] 'binlog_format' is deprecated and will be removed in a future release.
2024-04-28T14:02:01.081484Z 0 [Warning] [MY-011068] [Server] The syntax 'log_slave_updates' is deprecated and will be removed in a future release. Please use log_replica_updates instead.
2024-04-28T14:02:01.081503Z 0 [Warning] [MY-011068] [Server] The syntax 'skip_slave_start' is deprecated and will be removed in a future release. Please use skip_replica_start instead.
2024-04-28T14:02:01.081737Z 0 [System] [MY-010116] [Server] /opt/bitnami/mysql/bin/mysqld (mysqld 8.3.0) starting as process 73
2024-04-28T14:02:01.095011Z 1 [System] [MY-013576] [InnoDB] InnoDB initialization has started.
2024-04-28T14:02:01.706274Z 1 [System] [MY-013577] [InnoDB] InnoDB initialization has ended.
mysqld: Table 'mysql.plugin' doesn't exist
2024-04-28T14:02:01.966420Z 0 [ERROR] [MY-010735] [Server] Could not open the mysql.plugin table. Please perform the MySQL upgrade procedure.
2024-04-28T14:02:01.967294Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-28T14:02:01.967811Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-28T14:02:01.968321Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-28T14:02:01.969123Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-28T14:02:01.969534Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-28T14:02:01.969992Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-28T14:02:01.970453Z 0 [Warning] [MY-010441] [Server] Failed to open optimizer cost constant tables
2024-04-28T14:02:02.078915Z 4 [System] [MY-013381] [Server] Server upgrade from '80300' to '80300' started.
2024-04-28T14:02:18.415184Z 4 [System] [MY-013381] [Server] Server upgrade from '80300' to '80300' completed.
2024-04-28T14:02:18.858110Z 0 [Warning] [MY-010068] [Server] CA certificate ca.pem is self signed.
2024-04-28T14:02:18.858205Z 0 [System] [MY-013602] [Server] Channel mysql_main configured to support TLS. Encrypted connections are now supported for this channel.
2024-04-28T14:02:18.881693Z 0 [ERROR] [MY-000067] [Server] unknown variable 'master-info-repository=TABLE'.
2024-04-28T14:02:18.883803Z 0 [ERROR] [MY-010119] [Server] Aborting
2024-04-28T14:02:27.805040Z 0 [System] [MY-010910] [Server] /opt/bitnami/mysql/bin/mysqld: Shutdown complete (mysqld 8.3.0)  Source distribution.
2024-04-28T14:02:27.805094Z 0 [System] [MY-015016] [Server] MySQL Server - end.
```

everything is same as the example but I have disable allow empty password and provided required passwords

**Comment by dgomezleon** [comment]
Hi @mhkarimi1383 ,

Did you add `mysql-master` as suggested above? If so, please share your docker-compose file.

**Comment by mhkarimi1383** [comment]
@dgomezleon 

Here is my compose file

```yaml
services:
  my-db-master:
    image: docker.io/bitnami/mysql:8.3.0
    container_name: my-db-master
    expose:
      - '3306'
    volumes:
      - '/vol/db:/bitnami/mysql/data' # mkdir -p /vol/db && chown -R 1001:1001 /vol/db
    environment:
      - MYSQL_REPLICATION_MODE=master
      - MYSQL_REPLICATION_USER=repl_user
      - MYSQL_USER=my_user
      - MYSQL_DATABASE=my_db
      - ALLOW_EMPTY_PASSWORD=no
      - MYSQL_PASSWORD=my_passwd
      - MYSQL_ROOT_PASSWORD=root_passwd
      - MYSQL_REPLICATION_PASSWORD=repl_passwd
    healthcheck:
      test: ['CMD', '/opt/bitnami/scripts/mysql/healthcheck.sh']
      interval: 15s
      timeout: 5s
      retries: 6

  my-db-slave:
    image: docker.io/bitnami/mysql:8.3.0
    container_name: my-db-slave
    expose:
      - '3306'
    depends_on:
      - my-db-master
    environment:
      - TZ=Asia/Tehran
      - MYSQL_REPLICATION_MODE=slave
      - MYSQL_REPLICATION_USER=repl_user
      - MYSQL_USER=my_user
      - MYSQL_DATABASE=my_db
      - MYSQL_MASTER_HOST=my-db-master
      - MYSQL_MASTER_PORT_NUMBER=3306
      - MYSQL_MASTER_ROOT_PASSWORD=root_passwd
      - ALLOW_EMPTY_PASSWORD=no
      - MYSQL_PASSWORD=my_passwd
      - BITNAMI_DEBUG=true
      - MYSQL_REPLICATION_PASSWORD=repl_passwd
      # In case of missing binary files on master, use `true` to reset those binary files. Creating a previous backup is recommended.
      - MYSQL_REPLICATION_SLAVE_DUMP=false
    healthcheck:
      test: ['CMD', '/opt/bitnami/scripts/mysql/healthcheck.sh']
      interval: 15s
      timeout: 5s
      retries: 6
```

**Comment by mhkarimi1383** [comment]
I think this image is using a deprecated parameter

**Comment by dgomezleon** [comment]
Hi @mhkarimi1383 ,

Thanks for the info. I was able to reproduce the issue. This is the [cause](https://dev.mysql.com/doc/relnotes/mysql/8.3/en/news-8-3-0.html#:~:text=%2D%2Drelay%2Dlog%2Dinfo%2Dfile%2C%20%2D%2Drelay%2Dlog%2Dinfo%2Drepository%2C%20%2D%2Dmaster%2Dinfo%2Dfile%2C%20%2D%2Dmaster%2Dinfo%2Drepository%3A%20The%20use%20of%20files%20for%20the%20applier%20metadata%20repository%20and%20the%20connection%20metadata%20repository%20has%20been%20superseded%20by%20crash%2Dsafe%20tables%2C%20and%20is%20no%20longer%20supported.%20See%20Replication%20Metadata%20Repositories.).

I will release a new version to fix it ASAP.

**Comment by dgomezleon** [comment]
It should be solved now (`8.3.0-debian-12-r9`).
